Modules/_posixsubprocess.c

```diff
@@ -685,9 +685,16 @@ do_fork_exec(char *const exec_array[],
         assert(preexec_fn == Py_None);
 
         pid = vfork();
+        if (pid == -1) {
+            /* If vfork() fails, fall back to using fork(). When it isn't
+             * allowed from a process (such as pid==1 init), vfork can return
+             * -1 with errno EINVAL. https://bugs.python.org/issue47151. */
+            goto fallback_to_fork;
```

Comment:
If possible, I prefer to avoid goto. Here you can just use `pid = fork();`, no?

Reply:
heh, yes! thanks, done. :)